src/compiler/types.ts

```diff
@@ -6351,6 +6351,7 @@ namespace ts {
     }
 
     export const enum Extension {
+        Extensionless = "",

```

Comment:
No. Please do not add this here

Reply:
Ok, thanks for the context. I don't feel equipped to do that testing & detailed review as an outside contributor. Unless directed otherwise I'll leave this PR open for a while and it's still sitting in a few months I'll plan on closing it to keep the queue small.

If that happens and I'm not wrong, maybe it'd be good to mark the issue as an `effort: moderate` or `effort: difficult`? I really don't see a way to expand the inner workings of the module resolution logic to allow arbitrary extensions.

Reply:
yes this seems like a bigger change and would need more effort.. i am marking the issue accordingly.